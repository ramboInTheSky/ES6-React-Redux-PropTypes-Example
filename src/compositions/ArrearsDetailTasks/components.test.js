import React from 'react';
import { shallow } from 'enzyme';

import {
  Task,
  TaskBody,
  TaskDescription,
  TaskLabel,
  Label,
  TaskValue,
  TaskLink,
  Wrapper,
  H3,
  OverdueLabel,
  ButtonsWrapper,
} from './components';

const props = {
  theme: {
    colors: {
      secondaryLight: 'grey',
      support: { two: 'red', one: 'blue', six: 'yellow' },
    },
  },
};

describe('<Task />', () => {
  it('should render correctly', () => {
    expect(shallow(<Task {...props} />)).toMatchSnapshot();
  });
});

describe('<TaskBody />', () => {
  it('should render correctly', () => {
    expect(shallow(<TaskBody {...props} />)).toMatchSnapshot();
  });
});

describe('<TaskDescription />', () => {
  it('should render correctly', () => {
    expect(shallow(<TaskDescription {...props} />)).toMatchSnapshot();
  });
});

describe('<TaskLink />', () => {
  it('should render correctly', () => {
    expect(shallow(<TaskLink {...props} />)).toMatchSnapshot();
  });
});

describe('<H3 />', () => {
  it('should render correctly', () => {
    expect(shallow(<H3 />)).toMatchSnapshot();
  });
});

describe('<OverdueLabel />', () => {
  it('should render correctly', () => {
    expect(shallow(<OverdueLabel {...props} />)).toMatchSnapshot();
  });
});

describe('<ButtonsWrapper />', () => {
  it('should render correctly', () => {
    expect(shallow(<ButtonsWrapper {...props} />)).toMatchSnapshot();
  });
});

describe('<TaskLabel />', () => {
  it('should render correctly', () => {
    expect(shallow(<TaskLabel {...props} />)).toMatchSnapshot();
  });
});

describe('<Label />', () => {
  it('should render correctly', () => {
    expect(shallow(<Label {...props} />)).toMatchSnapshot();
  });
});

describe('<TaskValue />', () => {
  it('should render correctly', () => {
    expect(shallow(<TaskValue {...props} />)).toMatchSnapshot();
  });
});

describe('<Wrapper />', () => {
  it('should render correctly', () => {
    expect(shallow(<Wrapper />)).toMatchSnapshot();
  });
});
