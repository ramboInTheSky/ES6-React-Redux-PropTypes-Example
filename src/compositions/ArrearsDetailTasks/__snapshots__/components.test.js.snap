// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonsWrapper /> should render correctly 1`] = `
<div
  className="sc-iGPElx hwuxuB"
/>
`;

exports[`<H3 /> should render correctly 1`] = `
<styled.h3
  className="sc-etwtAo dnVaOU"
/>
`;

exports[`<Label /> should render correctly 1`] = `
<div
  className="sc-RcBXQ ftEQLC"
/>
`;

exports[`<OverdueLabel /> should render correctly 1`] = `
<span
  className="sc-jXQZqI eeVkhz"
/>
`;

exports[`<Task /> should render correctly 1`] = `
<div
  className="sc-fcdeBU cTbQBY"
/>
`;

exports[`<TaskBody /> should render correctly 1`] = `
<div
  className="sc-gmeYpB dgOrOU"
/>
`;

exports[`<TaskDescription /> should render correctly 1`] = `
<div
  className="sc-kZmsYB caSrI"
/>
`;

exports[`<TaskLabel /> should render correctly 1`] = `
<span
  className="sc-iSDuPN cgbHBY"
/>
`;

exports[`<TaskLink /> should render correctly 1`] = `
<WithTheme(Button)
  className="sc-fQejPQ kTzHFK"
  theme={
    Object {
      "colors": Object {
        "secondaryLight": "grey",
        "support": Object {
          "one": "blue",
          "six": "yellow",
          "two": "red",
        },
      },
    }
  }
/>
`;

exports[`<TaskValue /> should render correctly 1`] = `
<span
  className="sc-fZwumE lcIlyV"
/>
`;

exports[`<Wrapper /> should render correctly 1`] = `
<div
  className="sc-clNaTc jhyjGY"
/>
`;
